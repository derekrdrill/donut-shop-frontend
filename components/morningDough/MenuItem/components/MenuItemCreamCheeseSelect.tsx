import * as React from 'react';
import { Grid } from '@mui/material';

import MenuItemSelect from './MenuItemSelect';

import { FullMenuItem } from '../../Menu/assets/data/FULL_MENU';

type MenuItemCreamCheeseSelectProps = {
  menuItemData: FullMenuItem;
  selectedCreamCheese: string;
  setSelectedCreamCheese: React.Dispatch<React.SetStateAction<string>>;
};

const MenuItemCreamCheeseSelect = ({
  menuItemData,
  selectedCreamCheese,
  setSelectedCreamCheese,
}: MenuItemCreamCheeseSelectProps) =>
  menuItemData?.subCategory === 'bagels' ? (
    <Grid data-testid='cream-cheese-select' item xs={12} md={6} lg={4}>
      <MenuItemSelect
        placeholderText='Select your cream cheese'
        selectData={[
          { value: 'regular', text: 'Regular' },
          { value: 'chiveAndOnion', text: 'Chive and Onion' },
          { value: 'gardenVeggie', text: 'Garden Veggie' },
          { value: 'vegan', text: 'Vegan Butter' },
        ]}
        setValue={setSelectedCreamCheese}
        title='Cream Cheese'
        value={selectedCreamCheese}
      />
    </Grid>
  ) : (
    <React.Fragment />
  );

export default MenuItemCreamCheeseSelect;
