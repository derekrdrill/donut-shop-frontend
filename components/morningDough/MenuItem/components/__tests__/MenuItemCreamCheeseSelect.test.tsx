import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import MenuItemCreamCheeseSelect from '../MenuItemCreamCheeseSelect';

describe('MenuItemCreamCheeseSelect unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(
        <MenuItemCreamCheeseSelect
          menuItemData={{
            key: 'plainBagel',
            name: 'Plain Bagel',
            image: 'plainBagel.png',
            category: 'bread',
            subCategory: 'bagels',
            soda: false,
            bottled: false,
          }}
          selectedCreamCheese='Regular'
          setSelectedCreamCheese={jest.fn()}
        />,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders MenuItemCreamCheeseSelect correctly (cream-cheese-select)', async () => {
    render(
      <MenuItemCreamCheeseSelect
        menuItemData={{
          key: 'plainBagel',
          name: 'Plain Bagel',
          image: 'plainBagel.png',
          category: 'bread',
          subCategory: 'bagels',
          soda: false,
          bottled: false,
        }}
        selectedCreamCheese='Regular'
        setSelectedCreamCheese={jest.fn()}
      />,
    );

    const menuItemButterSelect = await screen.findAllByTestId('cream-cheese-select');

    expect(menuItemButterSelect[0]).toHaveClass('MuiGrid-root');
  });
});
