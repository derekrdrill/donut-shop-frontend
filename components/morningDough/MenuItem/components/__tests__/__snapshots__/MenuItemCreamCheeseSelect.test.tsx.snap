// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemCreamCheeseSelect unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
  data-testid="cream-cheese-select"
>
  <div
    className="MuiGrid-root MuiGrid-container css-1sjizby-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
      >
        Cream Cheese
      </label>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Without label"
          className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden={true}
          aria-invalid={false}
          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          tabIndex={-1}
          value="Regular"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
